solidity/scripts/compile.js

```diff
@@ -0,0 +1,79 @@
+const path = require('path');
+const fs = require('fs-extra');
+
+const {
+	contractsPath, buildPath, abisPath,
+	removeBuildOutput
+} = require('./_paths');
+
+const lastBuildTime = getLastBuildTime(buildPath, abisPath);
+console.log("Last Build Time:", lastBuildTime ? new Date(lastBuildTime) : "Never");
+
+var files = fs.readdirSync(contractsPath);
+
+const hasChanges = files
+	.map(f => path.resolve(contractsPath, f))
+	.map(p => fs.statSync(p).mtime > lastBuildTime)
+	.reduce((a, b) => a || b);
+if (hasChanges) {
+	removeBuildOutput();
+} else {
+	console.log(`No changes made to contracts since last build`);
+	return; // Stop executing
+}
+
```

Comment:
This is a cool function you made here to automatically compile the contract only when it needs to be recompiled. I'll want you to explain the functionality to me.